common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/ActionBottomSheetDialog.kt

```diff
@@ -0,0 +1,110 @@
+ * limitations under the License.
+ */
+
+package com.duckduckgo.mobile.android.ui.view.dialog
+
+import android.content.Context
+import android.view.LayoutInflater
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import com.duckduckgo.mobile.android.R
+import com.duckduckgo.mobile.android.databinding.BottomSheetActionBinding
+import com.duckduckgo.mobile.android.ui.view.show
+import com.google.android.material.bottomsheet.BottomSheetDialog
+
+class ActionBottomSheetDialog(builder: Builder) : BottomSheetDialog(builder.context, R.style.Widget_DuckDuckGo_BottomSheetDialog) {
+
+    abstract class EventListener {
+        open fun onBottomSheetShown() {}
+        open fun onBottomSheetDismissed() {}
+        open fun onPrimaryItemClicked() {}
+        open fun onSecondaryItemClicked() {}
+    }
+
+    internal class DefaultEventListener : EventListener()
+
+    private val binding: BottomSheetActionBinding = BottomSheetActionBinding.inflate(LayoutInflater.from(context))
+
+    init {
+        setContentView(binding.root)
+
+        setOnDismissListener { builder.listener.onBottomSheetDismissed() }
+        setOnShowListener { builder.listener.onBottomSheetShown() }
+        binding.actionBottomSheetDialogPrimaryItem.setOnClickListener { builder.listener.onPrimaryItemClicked() }
+        binding.actionBottomSheetDialogSecondaryItem.setOnClickListener { builder.listener.onSecondaryItemClicked() }
+
+        builder.titleText?.let {
+            binding.actionBottomSheetDialogTitle.text = it
+            binding.actionBottomSheetDialogTitle.show()
+        }
+
+        binding.actionBottomSheetDialogPrimaryItem.setPrimaryText(builder.primaryItemText)
+        builder.primaryItemIcon?.let { binding.actionBottomSheetDialogPrimaryItem.setLeadingIcon(it) }
+        builder.primaryItemTextColor?.let { binding.actionBottomSheetDialogPrimaryItem.setPrimaryTextColor(it) }
+
+        binding.actionBottomSheetDialogSecondaryItem.setPrimaryText(builder.secondaryItemText)
+        builder.secondaryItemIcon?.let { binding.actionBottomSheetDialogSecondaryItem.setLeadingIcon(it) }
+        builder.secondaryItemTextColor?.let { binding.actionBottomSheetDialogSecondaryItem.setPrimaryTextColor(it) }
+    }
+
+    class Builder(val context: Context) {
+        var listener: EventListener = DefaultEventListener()
+        var titleText: String? = null
+        var primaryItemText: String = ""
+        var primaryItemIcon: Int? = null
+        var primaryItemTextColor: Int? = null
+        var secondaryItemText: String = ""
+        var secondaryItemIcon: Int? = null
+        var secondaryItemTextColor: Int? = null
+
+        fun addEventListener(eventListener: EventListener): Builder {
+            listener = eventListener
+            return this
+        }
+
+        fun setTitle(title: String): Builder {
+            titleText = title
+            return this
+        }
+
+        fun onPrimaryItem(
```

Comment:
```suggestion
        fun setPrimaryItem(
```

The on prefix is preferred for things like listeners. Here we are setting the values of the component, so I think set makes more sense. What do you think?